### PR TITLE
AP_Filesystem: Change the judgment null of methods that return nullptr

### PR DESCRIPTION
**Pull request by muramura**
I found myself using NULL in the decision of a method that returns NULLPTR.
I believe there is a difference between NULL and NULLPTR.
I will change the decision value from NULL to NULLPTR.

### CODE REVIEW

File: libraries/AP_Filesystem/AP_Filesystem_FATFS.cpp [hunk 1]
```diff
@@ -405,7 +405,7 @@ int32_t AP_Filesystem_FATFS::read(int fd, void *buf, uint32_t count)
 
     // fileno_to_fatfs checks for fd out of bounds
     fh = fileno_to_fatfs(fd);
-    if ( fh == NULL ) {
+    if ( fh == nullptr ) {
```
Comment (hendjoshsr71) [inline libraries/AP_Filesystem/AP_Filesystem_FATFS.cpp:408]:
```suggestion
    if (fh == nullptr) {
```

File: libraries/AP_Filesystem/AP_Filesystem_FATFS.cpp [hunk 2]
```diff
@@ -450,7 +450,7 @@ int32_t AP_Filesystem_FATFS::write(int fd, const void *buf, uint32_t count)
...
```
Comment (hendjoshsr71) [inline libraries/AP_Filesystem/AP_Filesystem_FATFS.cpp:453]:
```suggestion
    if (fh == nullptr) {
```

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]
Looks fine.

But you missed a few in the file. You might as well do them all at once. (I got 17 when I looked, Lines 67-80, Line 107.)

If you wanted to, there are a few style fixups you could do on a few lines since you changed the lines anyways.

**Comment by muramura** [comment]
@hendjoshsr71 san
Thanks for the comment.
I agreed.
I've adapted the style.

**Review by hendjoshsr71 — COMMENTED** [review]
Sorry if I was confusing. I meant there are still others spots with `== NULL` you could replace in this file. Using VSCode and this search `== NULL`  I find the following still left. 
Line 67
Line 69
Line 74
Line 80
Line107
Line 161

**Comment by muramura** [comment]
@hendjoshsr71 sam
The lines you mentioned are as follows.
The element value of the table and the return value of the method are null.
In particular, the C language methods are NULL.
The ArduPilot uses the return value of the method to make a decision.

Line 67: file_table[] returns NULL.
Line 69: The calloc method returns NULL.
Line 74: The calloc method returns NULL.
Line 80: The malloc method returns NULL.
Line 107: file_table[] returns NULL.
Line 161: file_table[].fh returns NULL.

**Comment by hendjoshsr71** [comment]
I'm not a C++ expert, but I did come across this https://stackoverflow.com/questions/53532697/c-style-when-using-c-function-in-c-should-i-check-for-null-or-nullptr

Either seems fine and what you have here is better than before :) 

But lets ask @peterbarker since I've heard he is a resident C++ expert recently ;)

**Review by khancyr — COMMENTED** [review]
yep, that would be great to get rid of the NULL from the codebase.
On nice things, beside off some cpp compile time check is that it also remove a lot of noise on static analysers!

**Review by magicrub — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Review by hendjoshsr71 — APPROVED** [review]

**Pull request merged**